src/moonlink/src/storage/iceberg/rest_catalog.rs

```diff
@@ -0,0 +1,272 @@
+    ) -> IcebergResult<Vec<TableIdent>> {
+        todo!("list tables is not supported");
+    }
+
+    async fn update_namespace(
+        &self,
+        _namespace_ident: &NamespaceIdent,
+        _properties: HashMap<String, String>,
+    ) -> IcebergResult<()> {
+        todo!("Update namespace is not supported");
+    }
+
+    async fn create_table(
+        &self,
+        namespace_ident: &NamespaceIdent,
+        creation: TableCreation,
+    ) -> IcebergResult<Table> {
+        self.catalog.create_table(namespace_ident, creation).await
+    }
+
+    async fn load_table(&self, table_ident: &TableIdent) -> IcebergResult<Table> {
+        self.catalog.load_table(table_ident).await
+    }
+
+    async fn drop_table(&self, table: &TableIdent) -> IcebergResult<()> {
+        self.catalog.drop_table(table).await
+    }
+
+    async fn table_exists(&self, table: &TableIdent) -> IcebergResult<bool> {
+        self.catalog.table_exists(table).await
+    }
+
+    async fn rename_table(&self, _src: &TableIdent, _dest: &TableIdent) -> IcebergResult<()> {
+        todo!("rename table is not supported");
+    }
+
+    async fn update_table(&self, mut _commit: TableCommit) -> IcebergResult<Table> {
+        todo!("update table is not supported");
+    }
+
+    async fn register_table(
+        &self,
+        __table: &TableIdent,
+        _metadata_location: String,
+    ) -> IcebergResult<Table> {
+        todo!("register existing table is not supported")
+    }
+}
+
+#[cfg(feature = "storage-gcs")]
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use iceberg::spec::{NestedField, PrimitiveType, Schema, Type};
+    use reqwest::Client;
+    const REST_CATALOG_PROP_URI: &str = "uri";
+
+    async fn ensure_namespace(catalog: &IcebergRestCatalog, ns: &str) {
+        let ns_ident = NamespaceIdent::from_vec(vec![ns.to_string()]).unwrap();
+        let _ = catalog.create_namespace(&ns_ident, HashMap::new()).await;
+    }
+
+    #[tokio::test]
+    async fn test_create_table() {
+        let builder = IcebergRestCatalogBuilder::default().with_client(Client::new());
+        let catalog = builder
+            .load(
+                "test",
+                HashMap::from([(
+                    REST_CATALOG_PROP_URI.to_string(),
+                    "http://localhost:8181".to_string(),
+                )]),
+            )
+            .await;
+
+        let binding = catalog.unwrap();
+        ensure_namespace(&binding, "ns1").await;
+
+        let builder = IcebergRestCatalogBuilder::default().with_client(Client::new());
+        let catalog = builder
+            .load(
+                "test",
+                HashMap::from([(
+                    REST_CATALOG_PROP_URI.to_string(),
+                    "http://localhost:8181".to_string(),
+                )]),
+            )
+            .await;
+        assert!(catalog.is_ok());
+        let table_creation = TableCreation::builder()
+            .name("test1".to_string())
```

Comment:
I'm afraid it's very easy to get flaky test. rust runs all unit tests parallelly, unless you explicitly declare them to be serial. I would suggest using a random table name here.